src/Java.Interop.Tools.JavaTypeSystem/Adapters/JavaXmlApiExporter.cs

```diff
@@ -0,0 +1,357 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Xml;
+using Java.Interop.Tools.JavaTypeSystem.Models;
+
+namespace Java.Interop.Tools.JavaTypeSystem
+{
+	public static class JavaXmlApiExporter
+	{
+		public static void Save (JavaTypeCollection types, string xmlFile)
+		{
+			using (var writer = XmlWriter.Create (xmlFile, new XmlWriterSettings {
+				Encoding = new UTF8Encoding (false, true),
+				Indent = true,
+				OmitXmlDeclaration = true,
+			}))
+				Save (types, writer);
+		}
+
+		public static void Save (JavaTypeCollection types, XmlWriter writer)
+		{
+			writer.WriteStartElement ("api");
+
+			if (types.Platform.HasValue ())
+				writer.WriteAttributeString ("platform", types.Platform);
+
+			foreach (var pkg in types.Packages.Values) {
+
+				if (!pkg.Types.Any (t => !t.IsReferenceOnly))
+					continue;
+
+				writer.WriteStartElement ("package");
+				writer.WriteAttributeString ("name", pkg.Name);
+
+				if (pkg.PropertyBag.TryGetValue ("merge.SourceFile", out var source))
+					writer.WriteAttributeString ("merge.SourceFile", source);
+
+				if (!string.IsNullOrEmpty (pkg.JniName))
```

Comment:
We can "normalize" things here, and if `pkg.JniName` *is* null/empty -- for whatever reason -- then we can create one based on the `pkg.Name` value:

```csharp
writer.WriteAttributeString ("jni-name",
        !string.IsNullOrEmpty (pkg.JniName)
        ? pkg.JniName
        : pkg.Name.Replace (".", "/"));
```

Reply:
I think there are 2 cases when `pkg.JniName` is empty:
- Built-in types, which will never be output here
- Types that are not in a package, which will not have a `pkg.Name` either

It "cannot" be `null`, as we have it set to not-null.

Reply:
If it "cannot" be `null`, then why have a check which assumes that it *could* be null (or empty)?

Or is it instead that for "globally scoped" types, we want:

```xml
<package name="">
  <class name="TypeInGlobalPackage" … />
</package>
```

and not:

```xml
<package name="" jni-name="">
  <class name="TypeInGlobalPackage" … />
</package>
```